src/common/serac_types.hpp

```diff
@@ -203,6 +204,68 @@ struct BoundaryCondition {
   std::unique_ptr<mfem::HypreParMatrix> eliminated_matrix_entries;
 };
 
+/**
+ * Wraps a (currently iterative) system solver and handles the configuration of linear
+ * or nonlinear solvers
+ */
+class SystemSolver : public mfem::Solver {
+public:
+  // TODO: Eliminate this once a dependency injection approach is used for the solvers
+  SystemSolver() = default;
+  /**
+   * Constructs a new solver wrapper
+   * @param[in] comm The MPI communicator object
+   * @param[in] lin_params The parameters for the linear solver
+   * @param[in] nonlin_params The optional parameters for the optional nonlinear solver
+   * @see serac::LinearSolverParameters
+   * @see serac::NonlinearSolverParameters
+   */
+  SystemSolver(MPI_Comm comm, const LinearSolverParameters& lin_params,
+               const std::optional<NonlinearSolverParameters>& nonlin_params = std::nullopt);
+
+  /**
+   * Sets a preconditioner for the underlying linear solver object
+   * @param[in] prec The preconditioner, of which the object takes ownership
+   * @note The preconditioner must be moved into the call
+   * @code(.cpp)
+   * solver.setPreconditioner(std::move(prec));
+   * @endcode
+   */
+  void setPreconditioner(std::unique_ptr<mfem::Solver>&& prec)
```

Comment:
Any thoughts on whether the MFEM naming convention should be used here since the class implements the `mfem::Solver` interface?

Reply:
my vote is to follow mfem's convention, as it is really an mfem thing living in serac.

Reply:
I agree with @samuelpmishLLNL . However, I could go either way.